source/Engine.cpp

```diff
@@ -272,7 +272,7 @@ void Engine::Place()
 	ships.clear();
 	ai.ClearOrders();
 
-	EnterSystem();
+	EnterSystem(const_cast<System &>(*player.GetSystem()));
```

Comment:
What's the cast for?

Reply:
~~ask the compiler~~

I need system and not const system to be able to link and unlink systems since that takes a non cast System *

Reply:
That cast is a red flag, telling you and others, "this should be done someplace else."

Reply:
I'll make GameData do it internally then since it holds the objects.
That said I'm wondering, is it best to search for that system in GameData to have a non cost object or to cast it once in it?